appservice/src/TunnelProxy.ts

```diff
@@ -155,8 +155,8 @@ export class TunnelProxy {
         this._openSockets = [];
     }
 
-    public async startProxy(): Promise<void> {
-        await this.checkTunnelStatusWithRetry();
+    public async startProxy(portNumber: number): Promise<void> {
```

Comment:
`portNumber` and `this._port` are too similar in naming. I suggest you pass both ports in to the constructor and give them more unique names with documentation